### ISSUE TITLE
Code behaves differently, and stops

### ISSUE THREAD

**Issue by MANLW-Vestas** [modin-project/modin#4889]
By doing some searching, I found Modin and it caught my interest quite fast.
To test it, i made a script to see if it works better or worse that the ordinary Pandas. This worked very nicely with Ray on my local machine, but my machine isn’t powerful enough to get a good result, so I got my hands on a really powerful server to run it on. I have then copied my script and all dependencies over to the server, and running it there.
On my local machine, when I import Ray, and initializes it in the beginning, it just goes straight to the code itself.
But on the server, it behaves very differently, apparently. So when I run the script, it starts what I suspect is a virtual environment or localhost. After that, it gives me a lot of prints with the ‘UserWarning’ classification. After some time, it just fails saying that:

The raylet exited immediately because the Ray agent failed. The raylet fate shares with the agent. This can happen because the Ray agent was unexpectedly killed or failed. See dashboard_agent.log for the root cause.

I have absolutely not idea why it fails on my server and not on my local machine. They run the exact same code, with the exact same dependencies within an identical Pyenv. So I’m a little lost here, and honestly I don’t know if this is more related to Ray rather than Modin.

I hope someone can help me in some way

**Comment by MANLW-Vestas** [comment]
Yes of course! My laptop has a I7-10850H with 32GB RAM. I don't know the specific specs of the server, but I do know that I was allocated 64 cores and 128GB RAM, 

And here is the logs as well as the error message itself:

[dashboard.log](https://github.com/modin-project/modin/files/9450274/dashboard.log)
[dashboard_agent.log](https://github.com/modin-project/modin/files/9450281/dashboard_agent.log)

```
(pid=) [2022-08-30 07:22:31,087 E 15816 17980] (raylet.exe) agent_manager.cc:107: The raylet exited immediately because the Ray agent failed. The raylet fate shares with the agent. This can happen because the Ray agent was unexpectedly killed or failed. See `dashboard_agent.log` for the root cause.
Traceback (most recent call last):
  File "C:\Users\MANLW\Documents\testing\lpf_linking.py", line 374, in <module>
    matrix = generate_state_matrix(sup, exc, time_resolution="5s", do_interpolation=True)
  File "C:\Users\MANLW\Documents\testing\lpf_linking.py", line 229, in generate_state_matrix
    df = join_state_matrices(state_df_sv, state_df_ex)
  File "C:\Users\MANLW\Documents\testing\lpf_linking.py", line 158, in join_state_matrices
    df = pd.merge_ordered(df1, df2, how='outer')
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\logging\logger_function.py", line 65, in run_and_log
    return f(*args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\pandas\general.py", line 132, in merge_ordered
    right = to_pandas(right)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\utils.py", line 399, in to_pandas
    return modin_obj._to_pandas()
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\logging\logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\pandas\dataframe.py", line 2884, in _to_pandas
    return self._query_compiler.to_pandas()
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\logging\logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\storage_formats\pandas\query_compiler.py", line 259, in to_pandas
    return self._modin_frame.to_pandas()
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\logging\logger_metaclass.py", line 68, in log_wrap
    return method(*args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\dataframe\dataframe.py", line 115, in run_f_on_minimally_updated_metadata
    result = f(self, *args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\dataframe\dataframe.py", line 2835, in to_pandas
    df = self._partition_mgr_cls.to_pandas(self._partitions)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\partitioning\partition_manager.py", line 639, in to_pandas
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\partitioning\partition_manager.py", line 639, in <listcomp>
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\partitioning\partition_manager.py", line 639, in <listcomp>
    retrieved_objects = [[obj.to_pandas() for obj in part] for part in partitions]
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\dataframe\pandas\partitioning\partition.py", line 125, in to_pandas
    dataframe = self.get()
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\modin\core\execution\ray\implementations\pandas_on_ray\partitioning\partition.py", line 80, in get
    result = ray.get(self.oid)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\ray\_private\client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\MANLW\.env\cps_ml\lib\site-packages\ray\worker.py", line 1811, in get
    raise value
ray.exceptions.LocalRayletDiedError: The task's local raylet died. Check raylet.out for more information.

```

**Comment by mvashishtha** [comment]
@MANLW-Vestas thanks for sending that information. Could you also send the `raylet.out` log as the end of the error message says, as well as the `raylet.err`? Those [should be](https://docs.ray.io/en/latest/ray-observability/ray-logging.html#logging-directory-structure) in `/tmp/ray/session_latest/logs/` after you run your script.

Additionally, can you run either of the following scripts without error:

```python
import modin.pandas as pd

print(pd.DataFrame([3]))
```

```python
import ray

oid = ray.put(3)
print(ray.get(oid))
```

If you can run either of those scripts, then we can narrow the problem down to the usage of ray via your script in particular.

**Comment by MANLW-Vestas** [comment]
Here are the two raylet files. The 
[raylet.out.txt](https://github.com/modin-project/modin/files/9458594/raylet.out.txt)
[raylet.err.txt](https://github.com/modin-project/modin/files/9458595/raylet.err.txt)

And yes, I'm able to run both those tests without errrors. In the Modin test, it comes with the following:
```
UserWarning: Ray execution environment not yet initialized. Initializing...
To remove this warning, run the following python code before doing dataframe operations:

    import ray
    ray.init()

UserWarning: Distributing <class 'list'> object. This may take some time.
   0
0  3
```

While in the Ray test this is the output:
```
2022-08-31 06:14:18,135 INFO services.py:1456 -- View the Ray dashboard at http://127.0.0.1:8265
3
```

A thing I just noticed, is that in the ray test, it starts a localhost, and in the modin test, it recommends it, if I understand that message correctly.
But in my own script it also comes with that recommendation of using ray.init(), but it then, fails on the first pandas line, if it hasn't initialized

**Comment by pyrito** [comment]
@MANLW-Vestas it would help us a lot if you could post a minimum reproducible example of the problem you are facing so that we can triage the issue!

**Comment by MANLW-Vestas** [comment]
I have made a little script that produces the problem that I have. Hope this can be used.

The variable `n` represents a file that I have, and it's the same amount of entries
```python
import ray
ray.init() 
import random
import modin.pandas as pd
from datetime import datetime

def main():
    
    n = 676532

    matrix = pd.DataFrame({
        "timestamp": [datetime.fromtimestamp(i) for i in range(n)],
        1:  [random.getrandbits(1) for _ in range(n)],
        2:  [random.getrandbits(1) for _ in range(n)],
        3:  [random.getrandbits(1) for _ in range(n)],
        4:  [random.getrandbits(1) for _ in range(n)],
        5:  [random.getrandbits(1) for _ in range(n)],
        6:  [random.getrandbits(1) for _ in range(n)],
        7:  [random.getrandbits(1) for _ in range(n)],
        8:  [random.getrandbits(1) for _ in range(n)],
        9:  [random.getrandbits(1) for _ in range(n)],
        10: [random.getrandbits(1) for _ in range(n)],
        11: [random.getrandbits(1) for _ in range(n)],
        12: [random.getrandbits(1) for _ in range(n)],
        13: [random.getrandbits(1) for _ in range(n)],
        14: [random.getrandbits(1) for _ in range(n)],
        15: [random.getrandbits(1) for _ in range(n)],
        16: [random.getrandbits(1) for _ in range(n)],
        17: [random.getrandbits(1) for _ in range(n)],
        18: [random.getrandbits(1) for _ in range(n)],
        19: [random.getrandbits(1) for _ in range(n)],
        20: [random.getrandbits(1) for _ in range(n)],
        21: [random.getrandbits(1) for _ in range(n)],
        22: [random.getrandbits(1) for _ in range(n)],
        23: [random.getrandbits(1) for _ in range(n)],
        24: [random.getrandbits(1) for _ in range(n)],
        25: [random.getrandbits(1) for _ in range(n)],
        26: [random.getrandbits(1) for _ in range(n)],
        27: [random.getrandbits(1) for _ in range(n)],
        28: [random.getrandbits(1) for _ in range(n)],
        29: [random.getrandbits(1) for _ in range(n)],
        30: [random.getrandbits(1) for _ in range(n)],
        31: [random.getrandbits(1) for _ in range(n)],
        32: [random.getrandbits(1) for _ in range(n)],
        33: [random.getrandbits(1) for _ in range(n)],
        34: [random.getrandbits(1) for _ in range(n)],
        35: [random.getrandbits(1) for _ in range(n)],
        36: [random.getrandbits(1) for _ in range(n)],
        37: [random.getrandbits(1) for _ in range(n)],
        38: [random.getrandbits(1) for _ in range(n)],
        39: [random.getrandbits(1) for _ in range(n)],
        40: [random.getrandbits(1) for _ in range(n)],
        41: [random.getrandbits(1) for _ in range(n)],
        42: [random.getrandbits(1) for _ in range(n)],
        43: [random.getrandbits(1) for _ in range(n)],
        44: [random.getrandbits(1) for _ in range(n)],
        45: [random.getrandbits(1) for _ in range(n)],
        46: [random.getrandbits(1) for _ in range(n)],
        47: [random.getrandbits(1) for _ in range(n)],
        48: [random.getrandbits(1) for _ in range(n)],
        49: [random.getrandbits(1) for _ in range(n)],
    })

    matrix["timestamp"] = pd.to_datetime(list(matrix["timestamp"]), infer_datetime_format=True)
    matrix.set_index("timestamp", inplace=True)

    get_event_intervals(matrix)

def get_event_intervals(state_matrix: pd.DataFrame,) -> pd.DataFrame:
    
    state_matrix = state_matrix.sort_values(by = ['timestamp']).reset_index(drop = False)

    state_matrix.sort_values(by=["timestamp"], inplace=True)
    state_matrix['time_delta'] = (state_matrix['timestamp'] - state_matrix['timestamp'].shift(1)).astype('timedelta64[s]')
                         
    state_matrix['time_delta_acc'] = state_matrix['time_delta'].cumsum()
    state_matrix['time_delta_acc'] = state_matrix['time_delta_acc'].fillna(0.0)
    state_matrix['time_delta'] = state_matrix['time_delta'].fillna(method = 'bfill')

    event_intervals = state_matrix.melt(
        id_vars = ['timestamp', 'time_delta', 'time_delta_acc'],
        var_name = 'event_id', 
        value_name = 'state')
    
    event_intervals['event_id'] = event_intervals['event_id'].astype(str)

    event_intervals = event_intervals.sort_values(by=['event_id', 'timestamp']).reset_index(drop=True)
    event_intervals['interval_id'] = (
            (event_intervals['state'] != event_intervals['state'].shift(1))
            | (event_intervals['event_id'] != event_intervals['event_id'].shift(1))
    ).astype('int').cumsum()

    event_intervals = event_intervals.groupby(['event_id', 'state', 'interval_id']).agg(
        {'time_delta' : 'sum', 'time_delta_acc' : 'first', 'timestamp' : 'min'}).reset_index(drop = False)

    event_intervals.rename(
        columns = {'time_delta' : 'duration', 'time_delta_acc' : 'time_from_start', 'timestamp' : 'timestamp_min'},
        inplace = True)

    event_intervals['timestamp_max'] = event_intervals['timestamp_min'] + pd.to_timedelta(event_intervals['duration'], unit = 's')

    return event_intervals[[c for c in event_intervals.columns if c not in ['interval_id']]]

if __name__ == "__main__":
    main()
```

**Comment by mvashishtha** [comment]
I think the problem in @MANLW-Vestas 's script is `event_intervals['timestamp_max'] = event_intervals['timestamp_min'] + pd.to_timedelta(event_intervals['duration'], unit = 's')`. That's giving me `ValueError: Unable to coerce to Series, length must be 2: given 98` and `ValueError: Unable to coerce to Series, length must be 32: given 98`. That is #4967, but the root cause is #4966, which makes the right operand a `TimedeltaIndex` instead of a modin series. Once solve #4966, I think this script should be fixed, because adding series to series works without error. So I'm going to mark this as a duplicate of #4966.

@MANLW-Vestas we'll try to fix #4966 soon. Thank you for giving us a complete reproducer.

<details>
<summary>Ray task error</summary>

```
  File "/Users/maheshvashishtha/software_sources/modin/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py", line 450, in _apply_list_of_funcs
    partition = func(partition.copy(), *args, **kwargs)
  File "/Users/maheshvashishtha/software_sources/modin/modin/core/dataframe/algebra/binary.py", line 115, in <lambda>
    lambda df: func(df, other, *args, **kwargs),
  File "/Users/maheshvashishtha/opt/anaconda3/envs/modin-dev/lib/python3.10/site-packages/pandas/core/ops/__init__.py", line 434, in f
    self, other = align_method_FRAME(self, other, axis, flex=True, level=level)
  File "/Users/maheshvashishtha/opt/anaconda3/envs/modin-dev/lib/python3.10/site-packages/pandas/core/ops/__init__.py", line 282, in align_method_FRAME
    right = to_series(right)
  File "/Users/maheshvashishtha/opt/anaconda3/envs/modin-dev/lib/python3.10/site-packages/pandas/core/ops/__init__.py", line 233, in to_series
    raise ValueError(
ValueError: Unable to coerce to Series, length must be 2: given 98
```

</details>

**Comment by mvashishtha** [comment]
Duplicate of #4966
